account_invoice_tax_required/models/account_invoice.py

```diff
@@ -27,15 +13,15 @@ class AccountInvoice(models.Model):
     def test_invoice_line_tax(self):
         errors = []
         error_template = _("Invoice has a line with product %s with no taxes")
-        for invoice in self:
-            for invoice_line in invoice.invoice_line:
-                if not invoice_line.invoice_line_tax_id:
-                    error_string = error_template % (invoice_line.name)
-                    errors.append(error_string)
+        for invoice_line in self.mapped('invoice_line_ids'):
+            if not invoice_line.invoice_line_tax_ids:
+                error_string = error_template % (invoice_line.name)
+                errors.append(error_string)
         if errors:
             errors_full_string = ','.join(x for x in errors)
-            raise exceptions.Warning(_('No Taxes Defined!'),
-                                     errors_full_string)
+            raise exceptions.Warning(
+                _('No Taxes Defined!\n %s') % (errors_full_string)
```

Comment:
@sylvainvh I update this yesterday to prevent translation broken, and join errors with '\n'